astropy/table/serialize.py

```diff
@@ -230,12 +236,21 @@ def represent_mixins_as_columns(tbl, exclude_classes=()):
                                    exclude_classes=exclude_classes)
 
     # If no metadata was created then just return the original table.
-    if not mixin_cols:
-        return tbl
-
-    meta = deepcopy(tbl.meta)
-    meta['__serialized_columns__'] = mixin_cols
-    out = Table(new_cols, meta=meta, copy=False)
+    if mixin_cols:
+        meta = deepcopy(tbl.meta)
+        meta['__serialized_columns__'] = mixin_cols
+        out = Table(new_cols, meta=meta, copy=False)
+    else:
+        out = tbl
+
+    for col in out.itercols():
+        if not isinstance(col, Column) and col.__class__ not in exclude_classes:
```

Comment:
How can one reach this? Shouldn't `_represent_mixin_as_column` already have raised an error?
...
...
Ah, I now see your test. Could you add a comment that this catches columns for which `info` has not been set up right and therefore were not converted?

Or could the test be moved to right after/combined with the mixin-column conversion?

Reply:
Added code comment.